docs/architecture.md

```diff
@@ -0,0 +1,70 @@
+
+- Proxy call
+    - These are always POST calls due to the nature of how the HTTP spec works -- we want to send metadata to the endpoint and GETs are not used for this
+    - The call is made from the client with the target OpenShift Route path, request params, and the actual method you want to use
+- K8s call
+    - Powered by the [OpenShift SDK](https://github.com/openshift/dynamic-plugin-sdk)
+    - These are crafted with the k8s<Verb>Resource calls and makes use of a known model system
+        - Each call takes a `ResourceModel` and additional params like namespace (if applicable) & name (if applicable) and the SDK crafts a url to the endpoint
+        - Our server will take this call and proxy it onto the K8s API server
+    - See [SDK tidbits](./SDK.md)
+- Prometheus call
+    - Uses known prometheus endpoints to get data
+    - Somewhat deprecated -- it could probably be reworked to use the Proxy call endpoint, but serves as isolated functionality
+
+## Client Structure
+
+> **Note**: All content assumes you are looking at the `/frontend` folder.
+
+When building new client features, there are a few things worth noting about the current direction of the codebase.
+
+### Coding Structure
+
+> **Note**: Some sections do not fully comply -- there is a cleanup effort underway, do not let that dissuade you from the correct architecture choices
+
+- `/src/api` -- The current API
+    - This content should be isolated to the main effort to make the call
+    - There is some work here to clean up a little of the data, but effectively this folder should only talk about the very specific call structure needed to do a basic action (aka, less business logic - more functionality); eg, "Get Route" not "Get Notebook Route" or "Get DS Projects Notebook Route"
+        - If it's a cornerstone of data and has a specific core use-case, separate the logic; eg "Get Projects" & "Get DS Projects" are both in here
+- `/src/components` -- All generic components (not tied to a feature or functionality)
+    - eg. "IndentSection" - a component for indenting content
+    - eg. "Table" - a component for build tables
+- `/src/concepts` -- Sharable code and logic for multiple areas of the application; Read as "This is conceptually about X resource, doesn't care where it is mounted"
+    - eg. Reading project details / shared context
+    - eg. Shared conceptual logic -- "Components and utilities to parse and read Notebook resources"
+- `/src/pages` -- All specific view/route logic; Read as "This is tied to the nav item for this page"
+    - Should contain constants, utilities, hooks, and anything else that is specifically needed for that area of the application
+- `/src/utilities` -- All generic utilities and hook-utilities (not tied to a feature or functionality)
+
+> **Note**, if the folder was not mentioned above, it is deprecated and should be looked to not be added to anymore. 
+
+### Testing Structure
+
+TBD
```

Comment:
cc @Gkrumbach07 

I left this blank... did you want me to add my initial thoughts or did you want to do it after my PR? (or submit a PR against my branch in my fork)